src/process/mod.rs

```diff
@@ -1538,8 +1542,10 @@ impl std::iter::Iterator for ProcessesIter {
             match self.inner.next() {
                 Some(Ok(entry)) => {
                     if let Ok(pid) = i32::from_str(&entry.file_name().to_string_lossy()) {
-                        if let Ok(proc) = Process::new(pid) {
-                            break Some(Ok(proc));
+                        let process_root = PathBuf::from(&self.root).join(pid.to_string());
+                        match Process::new_with_root(process_root) {
+                            Ok(proc) => break Some(Ok(proc)),
+                            Err(e) => break Some(Err(e)),
```

Comment:
As it works now, the function will loop until a valid Process is found (Err is simply ignored). If you explicitly  handle the error, you can remove the `loop`, and the `break`s are also not useful anymore.

I prefer your wait of explicitly handling the error, but this changes the behavior a bit